src/dune_2_excel.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+from typing import List, Any
+from datetime import datetime
+
+import click
+from dotenv import load_dotenv
+from dune_client.client import DuneClient
+from dune_client.models import ResultsResponse
+from dune_client.query import Query
+from dune_client.types import QueryParameter
+import pandas as pd
+from tqdm import tqdm
+
+
+@click.command()
+@click.option("--query", "-q", multiple=True)
+@click.option("--start-date", "-s", type=click.DateTime(formats=["%Y-%m-%d"]))
+def main(query: List[str], start_date: datetime) -> Any:
+    """
+    Main function of the script
+    Args:
+        query: List of queries to fetch
+        start_date: start_date query parameter
+
+    Returns:
+
+    """
+    monthly_reporting(query, start_date)
+    return 0
+
+
+def store_results(results: List[ResultsResponse], start_date: datetime) -> None:
+    """
+    Store results into xlsx file
+    Args:
+        results: results to be stored
+        start_date: start_date query parameter
+    """
+    writer = pd.ExcelWriter(  # pylint: disable=abstract-class-instantiated
+        f'out/{"_".join([str(x.query_id) for x in results])}'
+        f'_{start_date.strftime("%Y-%m-%d")}.xlsx',
+        engine="xlsxwriter",
+    )
+    for result in results:
+        pd.DataFrame(pd.DataFrame(result.result.rows)).to_excel(
```

Comment:
Why two dataframe casts here?

Reply:
Also, I don't really see the need for involving pandas to turn a list of dictionaries into basically a CSV. What is pandas really doing for us here?

Reply:
```
src/dune_2_excel.py:47: error: Item "None" of "Optional[ExecutionResult]" has no attribute "rows"
```


Here (if you had used `get_rows`) you don't need to use result.result.rows anymore and this method swallows the Optional consideration logic.

Reply:
> Why two dataframe casts here?

That's a mistake I believe. Fixed it.

> Oh, I see its the to_excel method we need.

That's right.